js/reference.js

```diff
@@ -346,21 +346,82 @@ var ERMrest = (function(module) {
          * or errors (TBD).
          */
         create: function(data) {
+            var self = this;
             try {
-                // TODO
                 //  verify: data is not null, data has non empty tuple set
+                verify(data, "'data' must be specified");
+                verify(data.length > 0, "'data' must have at least one row to create");
+
+                var defer = module._q.defer();
+
                 //  get the defaults list for the referenced relation's table
-                //  get the data
+                var defaults = getDefaults();
+
+                //  get and format the data
+                // var formattedData = formatData(data);
+
+                // construct the uri
+                var uri = this._location.compactUri;
+                for (var i = 0; i < defaults.length; i++) {
+                    uri += (i === 0 ? "?defaults=" : ',') + module._fixedEncodeURIComponent(defaults[i]);
+                }
+
                 //  do the 'post' call
-                //  get the results from post (of course in a promise func)
-                //  make a page of tuples of the results (unless error)
-                //  new page will have a new reference (uri that filters on a disjunction of ids of these tuples)
-                //  resolve the promise, passing back the page
-                notimplemented();
+                module._http.post(uri, data).then(function(response) {
+                    //  new page will have a new reference (uri that filters on a disjunction of ids of these tuples)
+                    var uri = self._location.compactUri + '/',
+                        keyName;
+
+                    self._shortestKey.push(self._table.columns.get("c_int"));
```

Comment:
Oops, that was left over from me testing when the shortest key includes multiple columns.